hugegraph-cassandra/src/main/java/com/baidu/hugegraph/backend/store/cassandra/CassandraMetrics.java

```diff
@@ -50,36 +67,229 @@ public CassandraMetrics(CassandraSessionPool sessions, HugeConfig conf) {
+        String suffix = keyspace == null ? "*" : keyspace;
+        String name = humpToLine(metric + "_" + suffix);
+        // Aggregation of metrics for the whole host if keyspace=null
+        JmxTimerMBean value = (JmxTimerMBean) probe.getColumnFamilyMetric(
+                              keyspace, null, metric);
+        Map<String, Object> timerMap = InsertionOrderUtil.newMap();
+        timerMap.put("count", value.getCount());
+        timerMap.put("min", value.getMin());
+        timerMap.put("mean", value.getMean());
+        timerMap.put("max", value.getMax());
+        timerMap.put("stddev", value.getStdDev());
+        timerMap.put("p50", value.get50thPercentile());
+        timerMap.put("p75", value.get75thPercentile());
+        timerMap.put("p95", value.get95thPercentile());
+        timerMap.put("p98", value.get98thPercentile());
+        timerMap.put("p99", value.get99thPercentile());
+        timerMap.put("p999", value.get999thPercentile());
+        timerMap.put("duration_unit", value.getDurationUnit());
+        timerMap.put("mean_rate", value.getMeanRate());
+        timerMap.put("m15_rate", value.getFifteenMinuteRate());
+        timerMap.put("m5_rate", value.getFiveMinuteRate());
+        timerMap.put("m1_rate", value.getOneMinuteRate());
+        timerMap.put("rate_unit", value.getRateUnit());
+
+        metrics.put(name, timerMap);
+    }
+
+    protected static void appendCompactionMetrics(Map<String, Object> metrics,
+                                                  NodeProbe probe,
+                                                  String metric) {
+        // "CompletedTasks" => "compaction_completed_tasks"
+        String name = humpToLine("compaction" + metric);
+        Object value = probe.getCompactionMetric(metric);
+        if (value instanceof JmxCounterMBean) {
+            value = ((JmxCounterMBean) value).getCount();
+        }
+        metrics.put(name, value);
+    }
+
+    protected static void appendCacheMetrics(Map<String, Object> metrics,
+                                             NodeProbe probe,
+                                             String cacheType,
+                                             String metric) {
+        // "RowCache" + "Size" => "row_cache_size"
+        String name = humpToLine(cacheType + metric);
+        metrics.put(name, probe.getCacheMetric(cacheType, metric));
+    }
+
+    private static String humpToLine(String name) {
+        name = name.replaceAll("[A-Z]", "_$0").toLowerCase();
+        if (!name.isEmpty() && name.charAt(0) == '_') {
+            name = name.substring(1);
+        }
+        return name;
+    }
+
+    public Map<String, Object> compact() {
+        return this.executeAllHosts(this::compactHost);
+    }
+
+    private Object compactHost(String host) {
+        try (NodeProbe probe = this.newNodeProbe(host)) {
+            Compact compact = new Compact();
+            compact.execute(probe);
```

Comment:
compact all keyspaces?

Reply:
no, this api is for keyspace scope, need to fix